marshal.go

```diff
@@ -1716,6 +1724,169 @@ func unmarshalList(info TypeInfo, data []byte, value interface{}) error {
 	return unmarshalErrorf("can not unmarshal %s into %T. Accepted types: *slice, *array.", info, value)
 }
 
+func marshalVector(info VectorType, value interface{}) ([]byte, error) {
+	if value == nil {
+		return nil, nil
+	} else if _, ok := value.(unsetColumn); ok {
+		return nil, nil
+	}
+
+	rv := reflect.ValueOf(value)
+	t := rv.Type()
+	k := t.Kind()
+	if k == reflect.Slice && rv.IsNil() {
+		return nil, nil
+	}
+
+	switch k {
+	case reflect.Slice, reflect.Array:
+		buf := &bytes.Buffer{}
+		n := rv.Len()
+		if n != info.Dimensions {
+			return nil, marshalErrorf("expected vector with %d dimensions, received %d", info.Dimensions, n)
+		}
+
+		for i := 0; i < n; i++ {
+			item, err := Marshal(info.SubType, rv.Index(i).Interface())
+			if err != nil {
+				return nil, err
+			}
+			if isVectorVariableLengthType(info.SubType) {
+				writeUnsignedVInt(buf, uint64(len(item)))
+			}
+			buf.Write(item)
+		}
+		return buf.Bytes(), nil
+	}
+	return nil, marshalErrorf("can not marshal %T into %s", value, info)
+}
+
+func unmarshalVector(info VectorType, data []byte, value interface{}) error {
+	rv := reflect.ValueOf(value)
+	if rv.Kind() != reflect.Ptr {
+		return unmarshalErrorf("can not unmarshal into non-pointer %T", value)
+	}
+	rv = rv.Elem()
+	t := rv.Type()
+	k := t.Kind()
+	switch k {
+	case reflect.Slice, reflect.Array:
+		if data == nil {
+			if k == reflect.Array {
+				return unmarshalErrorf("unmarshal vector: can not store nil in array value")
+			}
+			if rv.IsNil() {
+				return nil
+			}
+			rv.Set(reflect.Zero(t))
+			return nil
+		}
+		if k == reflect.Array {
+			if rv.Len() != info.Dimensions {
+				return unmarshalErrorf("unmarshal vector: array of size %d cannot store vector of %d dimensions", rv.Len(), info.Dimensions)
+			}
+		} else {
+			rv.Set(reflect.MakeSlice(t, info.Dimensions, info.Dimensions))
+		}
+		elemSize := len(data) / info.Dimensions
+		for i := 0; i < info.Dimensions; i++ {
+			offset := 0
+			if isVectorVariableLengthType(info.SubType) {
+				m, p, err := readUnsignedVint(data, 0)
```

Comment:
The `start` parameter of the `readUnsignedVint` func seem to be redundunt. Could move inside the func. Or am I missing something?

Reply:
It is but I would leave it as is when data will not represent just unsigned Vint, but longer slice. But if that is a problem, I can change it.

Reply:
Hardcoding is kind of distracting. We can always add the parameter back when there's an actual requirement for it. I'd probably remove it if there is no use of it. 
@joao-r-reis thoughts?

Reply:
Done.

Reply:
> Hardcoding is kind of distracting. We can always add the parameter back when there's an actual requirement for it. I'd probably remove it if there is no use of it. @joao-r-reis, I’d appreciate your thoughts on this when you have a moment.

Agree